main/src/cgeo/geocaching/files/ImportGpxZipAttachmentThread.java

```diff
@@ -23,7 +25,18 @@ public ImportGpxZipAttachmentThread(final Uri uri, final ContentResolver content
 
     @Override
     protected InputStream getInputStream() throws IOException {
-        return contentResolver.openInputStream(uri);
+        try {
+            return contentResolver.openInputStream(uri);
+        } catch (final FileNotFoundException e) {
+            // for http links, we may need to download the content ourself, if it has no mime type announced by the browser
+            if (uri.toString().startsWith("http")) {
+                return Network.getResponseStream(Network.getRequest(uri.toString()));
+            }
+            // only log error for non-http URI
+            Log.e("GpxZip import cannot resolve " + uri);
```

Comment:
Those two lines should probably be located just above the `return null` (and the comment can be removed).